vortex-array/src/array/statistics.rs

```diff
@@ -62,6 +62,14 @@ impl<A: Array + ArrayImpl> Statistics for A {
             .set(stat, value);
     }
 
+    fn set_stats_iter(&self, stats_iter: StatsSetIntoIter) {
```

Comment:
Why not just take a StatsSet? This trait already has `stats_set() -> StatsSet`, so it's already aware of the type.